test/powershell/Modules/Microsoft.PowerShell.Security/FileCatalog.Tests.ps1

```diff
@@ -42,6 +42,25 @@ Describe "Test suite for NewFileCatalogAndTestFileCatalogCmdlets" -Tags "CI" {
 
     Context "NewAndTestCatalogTests PositiveTestCases when validation Succeeds" {
 
+        It "NewFileCatalogWithSingleFile with WhatIf" {
+
+            $sourcePath = Join-Path $testDataPath '\CatalogTestFile1.mof'
+            # use existant Path for the directory when .cat file name is not specified
+            $catalogPath = $testDataPath
+            try
+            {
+                $null = New-FileCatalog -Path $sourcePath -CatalogFilePath $catalogPath -WhatIf
+                $result = Test-Path -Path ($catalogPath + "\catalog.cat")
```

Comment:
I would introduce a variable `$catalogFile` or something like that and use the variable when trying to delete the file.

I would also probably *not* call `Remove-File` if `Test-Path` says the file isn't there.